contrib/apelscripts/condor_batch.sh

```diff
@@ -1,28 +1,32 @@
 #!/bin/bash
 
+fail () {
+    echo "$@" >&2
+    exit 1
+}
+
+safe_config_val () {
+    condor_ce_config_val $1 || fail "Failed to retrieve CE configuration value '$1'"
+}
+
 # Create a temporary accounting file name
 today=$(date -u --date='00:00:00 today' +%s)
 yesterday=$(date -u --date='00:00:00 yesterday' +%s)
-output=batch-$(date -u --date='yesterday' +%Y%m%d )-$(hostname -s)
 
-# Build the filter for the history command
-CONSTR="EnteredCurrentStatus >= $yesterday && EnteredCurrentStatus < $today && RemoteWallclockTime !=0"
+OUTPUT_DIR="$(condor_ce_config_val APEL_OUTPUT_DIR)"
+OUTPUT_FILE="$OUTPUT_DIR/batch-$(date -u --date='yesterday' +%Y%m%d )-$(hostname -s)"
 
-function die() { echo "$0 aborted when reading config"; exit 1; }
-trap die ERR;
-if [ $# -gt 0 ]; then
-  source $1
-fi
-trap '' ERR
+[[ -d $OUTPUT_DIR && -w $OUTPUT_DIR ]] || fail "Cannot write to $OUTPUT_DIR"
 
-[[ -z "$SCALING_ATTR" ]] && SCALING_ATTR=MachineAttrRalScaling0
-[[ -z "$BATCH_HOST" ]] && BATCH_HOST=`hostname`
-[[ -z "$OUTPUT_LOCATION" ]] && OUTPUT_LOCATION=/var/lib/condor-ce/apel/
+# Build the filter for the history command
+CONSTR="EnteredCurrentStatus >= $yesterday && EnteredCurrentStatus < $today && RemoteWallclockTime !=0"
 
-OUTPUT_FILE=$OUTPUT_LOCATION/$output
+HISTORY_EXTRA_ARGS='-format "\n" EMPTY'
+SCALING_ATTR=$(safe_config_val APEL_SCALING_ATTR)
```

Comment:
Ah, but it's not so easy because `safe_config_val` is run in a subshell, and the `exit 1` won't be propagated...

I can put together a suggestion...

Reply:
Wanna bet?

Reply:
Never mind. misunderstood the pr again.